ihatemoney/web.py

```diff
@@ -89,14 +91,24 @@ def admin():
     form = AdminAuthenticationForm()
     goto = request.args.get('goto', url_for('.home'))
     if request.method == "POST":
-        if form.validate():
-            if check_password_hash(current_app.config['ADMIN_PASSWORD'], form.admin_password.data):
-                session['is_admin'] = True
-                session.update()
-                return redirect(goto)
-            else:
-                msg = _("This admin password is not the right one")
+        client_ip = request.remote_addr
+        if login_throttler.is_login_allowed(client_ip):
+            if form.validate():
+                # Valid password
+                if (check_password_hash(current_app.config['ADMIN_PASSWORD'],
+                                        form.admin_password.data)):
+                    session['is_admin'] = True
+                    session.update()
+                    login_throttler.reset(client_ip)
+                    return redirect(goto)
+                # Invalid password
+                login_throttler.increment_attempts_counter(client_ip)
+                msg = _("This admin password is not the right one. Only %d attempts left." %
```

Comment:
[gettext](https://pythonhosted.org/Flask-Babel/#flask.ext.babel.gettext) can be passed the second argument directly to ease the work of translators.